core/src/parking_lot.rs

```diff
@@ -132,6 +135,9 @@ struct ThreadData {
 
     // ParkToken value set by the thread when it was parked
     park_token: Cell<ParkToken>,
+
+    // Resources
+    resources: UnsafeCell<Vec<usize>>,
```

Comment:
The name "resources" isn't very informative. You should make it clearer that this is for deadlock detection.